partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt

```diff
@@ -1089,6 +1089,9 @@ internal class PlanTyper(private val env: Env) {
          * Calculate output type of a scalar subquery.
          */
         private fun visitRexOpSubqueryScalar(subquery: Rex.Op.Subquery, cons: CompilerType): Rex {
+            if (cons.kind == Kind.DYNAMIC) {
```

Comment:
Why was this check needed?

Reply:
In the following lines, `cons.fields` is called -- which is not relevant to a dynamic type, causing an exception to be thrown.